src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/EditorConfigNamingStyleParser.NamingStyle.cs

```diff
@@ -0,0 +1,110 @@
+
+            namingStyle = new NamingStyle()
+            {
+                Name = namingStyleTitle,
+                Prefix = requiredPrefix,
+                Suffix = requiredSuffix,
+                WordSeparator = wordSeparator,
+                CapitalizationScheme = capitalization
+            };
+            return true;
+        }
+
+        private static bool TryGetNamingStyleTitle(
+            string namingRuleName,
+            IReadOnlyDictionary<string, object> conventionsDictionary,
+            out string namingStyleName)
+        {
+            object result;
+            if (conventionsDictionary.TryGetValue($"dotnet_naming_rule.{namingRuleName}.style", out result))
+            {
+                namingStyleName = (string)result;
+                return true;
+            }
+
+            namingStyleName = null;
+            return false;
+        }
+
+        private static string GetNamingRequiredPrefix(string namingStyleName, IReadOnlyDictionary<string, object> conventionsDictionary)
+        {
+            object result;
+            if (conventionsDictionary.TryGetValue($"dotnet_naming_style.{namingStyleName}.required_prefix", out result))
+            {
+                return (string)result;
+            }
+
+            return string.Empty;
+        }
+
+        private static string GetNamingRequiredSuffix(string namingStyleName, IReadOnlyDictionary<string, object> conventionsDictionary)
+        {
+            object result;
+            if (conventionsDictionary.TryGetValue($"dotnet_naming_style.{namingStyleName}.required_suffix", out result))
+            {
+                return (string)result;
+            }
+
+            return string.Empty;
+        }
+
+        private static string GetNamingWordSeparator(string namingStyleName, IReadOnlyDictionary<string, object> conventionsDictionary)
+        {
+            object result;
+            if (conventionsDictionary.TryGetValue($"dotnet_naming_style.{namingStyleName}.word_separator", out result))
+            {
+                return (string)result;
+            }
+
+            return string.Empty;
+        }
+
+        private static Capitalization GetNamingCapitalization(string namingStyleName, IReadOnlyDictionary<string, object> conventionsDictionary)
+        {
+            object result;
+            if (conventionsDictionary.TryGetValue($"dotnet_naming_style.{namingStyleName}.capitalization", out result))
+            {
+                return ParseCapitalizationScheme((string)result);
+            }
+
+            return default(Capitalization);
+        }
+
+        private static Capitalization ParseCapitalizationScheme(string namingStyleCapitalization)
+        {
+            switch (namingStyleCapitalization)
+            {
+                case "pascal_case": return Capitalization.PascalCase;
+                case "camel_case": return Capitalization.CamelCase;
+                case "first_word_upper": return Capitalization.FirstUpper;
```

Comment:
should this be Capitalization.FirstWordUpper?   or shoudl the string be "first_upper"?